google/cloud/pubsub/internal/streaming_subscription_batch_source.cc

```diff
@@ -236,14 +236,16 @@ void StreamingSubscriptionBatchSource::ExtendLeases(
   }
   lk.unlock();
   for (auto& r : split) {
+    Span modack_span = callback_->StartModackSpan(request);
```

Comment:
Why are we returning a `Span`? Shouldn't the tracing layer of `callback_` (which is `cb`) inside the lambda hold the `Span`?

Reply:
The alternative could be storing the CQ maybe in the default batch callback and adding tracing around that. Still thinking about what that would look like

Reply:
We talked offline. I was missing that it is difficult to associate the `StartModackSpan` with an `EndModackSpan`. Returning the `Span` was a way to connect the two.

Apparently @devbww made comments to this effect, but I didn't read them.